src/test/java/jchess/ruleengine/DefaultRuleProcessorTest2_2PlayerBoard.java

```diff
@@ -795,7 +797,177 @@ void testTryFindCandidateMovesForFileAndRankStrategy_MOVE_NORTHEAST_BY_INF() {
 			assertEquals(oExpectedPositionInMap, oActualPositionInMap);
 		}
 	}
-
+	
+	@Test
+	void testCheckIfPlayerEndengered_defaultBoard() {
+		
+		//Check Player Endergered when the game start
+		
+		IPlayerAgent oPlayerAgent = m_oBoard.getPlayerAgent("P1");
+		
+		IPlayerAgent oExpected = null;
+		IPlayerAgent oActual = m_oRuleProcessor.tryCheckIfPlayerEndengered(m_oBoard, oPlayerAgent);
+		
+		assertEquals(oExpected,oActual);
+	}
+	
+	@Test
+	void testCheckIfPlayerEndengered_RivalPawn_King_checkPosition()
```

Comment:
I think in this case you use DefaultRuleEngine. Please move it to its own test class.